pkg/training/trainer_ray.go

```diff
@@ -0,0 +1,373 @@
+// Copyright 2018 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package training
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/kubeflow/arena/pkg/apis/config"
+	"github.com/kubeflow/arena/pkg/apis/types"
+	"github.com/kubeflow/arena/pkg/apis/utils"
+	"github.com/kubeflow/arena/pkg/k8saccesser"
+	"github.com/ray-project/kuberay/ray-operator/pkg/client/clientset/versioned"
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+)
+
+const (
+	// rayjob-operator added labels for pods and servers.
+	rayClusterNodeType = "ray.io/node-type"
+	labelRayJobName    = "job-name"
+)
+
+// RayJob Information
+type RayJob struct {
+	*BasicJobInfo
+	rayJob       *rayv1.RayJob
+	pods         []*v1.Pod // all the pods including statefulset and job
+	chiefPod     *v1.Pod   // the head pod
+	requestedGPU int64
+	allocatedGPU int64
+	trainerType  types.TrainingJobType // return trainer type: rayjob
+}
+
```

Comment:
```suggestion
// RayJob implements the TrainingJob interface.
var _ TrainingJob = &RayJob{}
```

It would be better to add the above code to ensure that `RayJob` implements the `TrainingJob` interface.